### PR TITLE
OTBR support

### PR DESCRIPTION
**Pull request by krish2718**
Add an overlay for Wi-Fi thread combo, useful for the OTBR [1].
    
    OpenThread interface is the first interface though its not default, so,
    we need to explicitly initialize the network stack to use Wi-Fi
    interface else DHCPv4 chooses OpenThread and it will drop the packets as
    it only supports IPv6.
    
    [1] - https://openthread.io/guides/border-router

### CODE REVIEW

**Review by sr1dh48r — APPROVED** [review]

**Review by jciupis — CHANGES_REQUESTED** [review]
I have a couple of comments that I'd like to see addressed before this PR is merged.

1. I pulled the code from the PR and built the `wifi/shell` sample for nRF7002 DK using the following command:
```
west built -p -b nrf7002dk_nrf5340_cpuapp -- -DOVERLAY_CONFIG=./overlay-otbr.conf
```
It crashes almost immediately after boot. I'm not ruling out some hardware issue on my side, but so far it hasn't worked out of the box for me - let's synchronize directly on this.

2. I'm missing documentation of this new overlay. The name `overlay-otbr` is rather cryptic and its purpose is not documented anywhere. Could you please add, for example, a section in the sample's `README.rst`?
3. It seems to me that the commits in this PR are not bisectable. First the overlay is added, then the manifest update that should allow it to work is added, and then some fix for the sample is added. Could you please look into the commit history and restructure it to make it clean?

File: samples/wifi/shell/src/wifi_main.c [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** @file
+ * @brief WiFi shell sample main function
  */
 
 #include <zephyr/sys/printk.h>
 #include <nrfx_clock.h>
+#ifdef CONFIG_NET_L2_OPENTHREAD
```
Comment (sachinthegreen) [inline samples/wifi/shell/src/wifi_main.c:13]: I guess we can remove this conditional flag...i.e. make this applicable by default.

File: samples/wifi/shell/src/wifi_main.c [hunk 2]
```diff
@@ -15,4 +23,14 @@ void main(void)
 			       NRF_CLOCK_HFCLK_DIV_1);
 #endif
 	printk("Starting %s with CPU frequency: %d MHz\n", CONFIG_BOARD, SystemCoreClock/MHZ(1));
+
+#ifdef CONFIG_NET_L2_OPENTHREAD
```
Comment (sachinthegreen) [inline samples/wifi/shell/src/wifi_main.c:27]: I guess we can remove this conditional flag...i.e. make this applicable by default.

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

**Comment by krish2718** [comment]
> 1. I pulled the code from the PR and built the `wifi/shell` sample for nRF7002 DK using the following command:
> 
> ```
> west built -p -b nrf7002dk_nrf5340_cpuapp -- -DOVERLAY_CONFIG=./overlay-otbr.conf
> ```
> 
> It crashes almost immediately after boot. I'm not ruling out some hardware issue on my side, but so far it hasn't worked out of the box for me - let's synchronize directly on this.
As discussed offline this is not a crash rather an assert, it was because of mismatched build, compiling with `-DCONFIG_NRF700X_REV_A=y` works. But nonetheless it shouldn't crash, but the `net_if_up` mandatorily expectes MAC address even if the interface is failed to bring (for nrf7002 l2->enable is not defined). @rlubos is this expected?
> 
> 2. I'm missing documentation of this new overlay. The name `overlay-otbr` is rather cryptic and its purpose is not documented anywhere. Could you please add, for example, a section in the sample's `README.rst`?

Renamed the overlay to openthread (easier to fix this than write docs :) ).

> 3. It seems to me that the commits in this PR are not bisectable. First the overlay is added, then the manifest update that should allow it to work is added, and then some fix for the sample is added. Could you please look into the commit history and restructure it to make it clean?

LGTM now, please check.

**Comment by sachinthegreen** [comment]
> I have a couple of comments that I'd like to see addressed before this PR is merged.
> 
> 1. I pulled the code from the PR and built the `wifi/shell` sample for nRF7002 DK using the following command:
> 
> ```
> west built -p -b nrf7002dk_nrf5340_cpuapp -- -DOVERLAY_CONFIG=./overlay-otbr.conf
> ```
> 
> It crashes almost immediately after boot. I'm not ruling out some hardware issue on my side, but so far it hasn't worked out of the box for me - let's synchronize directly on this.
> 
> 2. I'm missing documentation of this new overlay. The name `overlay-otbr` is rather cryptic and its purpose is not documented anywhere. Could you please add, for example, a section in the sample's `README.rst`?
> 3. It seems to me that the commits in this PR are not bisectable. First the overlay is added, then the manifest update that should allow it to work is added, and then some fix for the sample is added. Could you please look into the commit history and restructure it to make it clean?



> 



> 

@jciupis : If you are ok with these resolutions, could you please give your upvote so that this PR can be merged?

**Review by jciupis — APPROVED** [review]
LGTM, sorry that I kept you waiting @sachinthegreen @krish2718 :)

**Pull request merged**